src/routes/transactions/mappers/common/transaction-info.mapper.ts

```diff
@@ -65,16 +65,15 @@ export class MultisigTransactionInfoMapper {
     const dataSize =
       dataByteLength >= 2 ? Math.floor((dataByteLength - 2) / 2) : 0;
 
-    const richDecodedInfo = this.isHumanDescriptionEnabled
+    const richDecodedInfo = this.isRichFragmentsEnabled
       ? await this.humanDescriptionMapper.mapRichDecodedInfo(
           transaction,
           chainId,
         )
       : null;
 
-    const humanDescription = this.isHumanDescriptionEnabled
-      ? this.humanDescriptionMapper.mapHumanDescription(richDecodedInfo)
-      : null;
+    const humanDescription =
+      this.humanDescriptionMapper.mapHumanDescription(richDecodedInfo);
```

Comment:
This takes `richDecodedInfo` as a parameter which would be `null` if the feature flag is disabled. We could pass `null` to the mapper further down in that case instead.

Reply:
Good catch! However this can complicate things as if we set it to `null` afterwards we need to assume that the mappers will never use the `richDecodedInfo` for other parts of the response (excluding `richDecodedInfo`) as it's the case with `humanDescription`.

Setting it to `null` afterwards would be the easiest solution but not sure if it'd be robust compared to performing the check on the final payloads (more occurrences 🤔 )

Reply:
I'm not sure I'm following the logic.

If `this.isRichFragmentsEnabled` is `false`, then `richDecodedInfo` is `null`, and per https://github.com/safe-global/safe-client-gateway/blob/55f36a0b0c54b9d2ee30892c5555c38ed7dd2eaf/src/routes/transactions/mappers/common/human-description.mapper.ts#L82 `humanDescription` will be also `null`. So both `richDecodedInfo` and `humanDescription` will be `null` if `this.isRichFragmentsEnabled` is `false`, right?

If I'm not wrong both variables still depend on the same FF, so no decoupling between features is being made, is that correct?

Reply:
We can additionally set the property to `undefined` afterwards as `undefined` properties would not be rendered on the API level.

@usame-algan on the client side do we expect the key `richDecodedInfo` to be present?

Reply:
I just checked and we don't rely on the existence of the `richDecodedInfo` in safe-wallet-web so it can be safely set to be `undefined` 👍

Reply:
Got it, thanks for explaining @fmrsabino 👍🏻